core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1900,7 +1900,17 @@ object SparkContext extends Logging {
 
   private[spark] val SPARK_JOB_INTERRUPT_ON_CANCEL = "spark.job.interruptOnCancel"
 
-  private[spark] val DRIVER_IDENTIFIER = "<driver>"
+  /**
+   * Executor id for the driver.  In earlier versions of Spark, this was `<driver>`, but this was
+   * changed to `driver` because the angle brackets caused escaping issues in URLs and XML (see
+   * SPARK-6716 for more details).
+   */
+  private[spark] val DRIVER_IDENTIFIER = "driver"
+
+  /**
+   * Legacy version of DRIVER_IDENTIFIER, retained for use in backwards-compatibility tests.
```

Comment:
This could be interpreted as implying it is only needed for tests, however, this is actually legitimately needed for reading old event logs. It might make more sense to just say it's needed for backwards compatibility.